index.js

```diff
@@ -80,6 +82,22 @@ function loggingFunction(options, log, tries) {
 
     if (!syslogConnectionSingleton.connection && !syslogConnectionSingleton.connecting) {
         syslogConnectionSingleton.connecting = true;
+        if (options.useUdpSyslog) {
+        	var client = dgram.createSocket('udp4');
+        	syslogConnectionSingleton.connection = {
+        		write: function (msg) {
+        			client.send(msg, 0, msg.length, options.port, options.host, function (err) {
+        				if (err && err !== 0) {
+        					cleanupConnection(err, 'error');
+        					retryLogic(loggingFunction.bind(this, options, log), tries);
+        				}
```

Comment:
At this point, maybe we want an else block to set `syslogConnectionSingleton.connecting = false`? otherwise the logMessage function wont ever be called subsequently.